frontend/src/pages/[user]/[repo]/index.tsx

```diff
@@ -79,53 +80,47 @@ RepoPage.getInitialProps = async (ctx: NextPageContext): Promise<RepoPageProps>
 }
 
 interface ProjectGridData {
-  swrFallback: Record<string, Array<unknown>>
+  swrFallback: Record<string, Array<Project>>
   gridProps: ProjectGridProps
 }
 
 interface ProjectGridProps {
   parentProject: string
-  searchArgs: SearchArgs
+  searchParams: SearchFetcherParams
 }
 
-const ProjectGrid = ({ parentProject, searchArgs }: ProjectGridProps) => {
-  const { data: projects } = useSWR(searchArgs, fetchSearch, {
+const PageContent = ({ parentProject, searchParams }: ProjectGridProps) => {
+  const { data: projects } = useSWR(searchParams, searchFetcher, {
     refreshInterval: 1000,
   })
   const isProcessing = projects.length === 0
+
   return (
     <Page title={parentProject}>
       <h1>{parentProject}</h1>
-      <div>
-        {isProcessing
-          ? 'Processing repository...'
-          : projects.map(project => <ProjectCard {...project} key={project.id} />)}
-      </div>
+      {isProcessing ? (
+        <p>Processing repository...</p>
+      ) : (
+        <ProjectCardGrid projects={projects} />
+      )}
     </Page>
   )
 }
 
-type SearchArgs = [string, SearchParams]
-
-const fetchSearch = async (...args: SearchArgs) => {
-  const searchResult = await meiliIndex.search(...args)
-  return searchResult.hits
-}
-
 const getGridData = async (
   username: string,
   repoName: string,
 ): Promise<ProjectGridData> => {
   const repoFullName = `${username}/${repoName}`
   const repo = await getRepo(repoFullName)
-  const searchArgs: SearchArgs = ['*', { filter: `repoId = ${repo.id}` }]
-  const hits = await fetchSearch(...searchArgs)
+  const searchParams = { filter: `repoId = ${repo.id}` }
```

Comment:
```suggestion
  const searchParams = { filter: `repoId = ${repo.id}`, limit: 1000 }
```

Just to make our limit explicit here. The default is 12 so I fell like it's actually possible to hit that, whereas 1000 doesn't seem likely.